### PR TITLE
support numerical precision and env_mat

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
- change of numerical precision is supported
- environment matrix is supported.

### CODE REVIEW

File: deepmd_utils/model_format/network.py [hunk 1]
```diff
@@ -198,6 +196,7 @@ def deserialize(cls, data: dict) -> "NativeLayer":
             idt=data["@variables"].get("idt", None),
             activation_function=data["activation_function"],
             resnet=data.get("resnet", False),
+            precision=data.get("precision", "default"),
```
Comment (njzjz) [inline deepmd_utils/model_format/network.py:199]: I propose
```suggestion
            precision=data["@variables"]["w"].dtype.name,
```

Comment (njzjz) [reply deepmd_utils/model_format/network.py:199]: Do we really need this key? I think we can get the type easily everywhere...

File: deepmd_utils/model_format/network.py [hunk 2]
```diff
@@ -156,10 +150,13 @@ def __init__(
         idt: Optional[np.ndarray] = None,
         activation_function: Optional[str] = None,
         resnet: bool = False,
+        precision: str = "default",
```
Comment (njzjz) [inline deepmd_utils/model_format/network.py:153]: Why not just set `float64` as the default.. In deepmd tf, `default` may have different meanings, but I think we need the exact meaning when storing the model.

**Review by njzjz — APPROVED** [review]

**Comment by njzjz** [comment]
I still have a question. Do we want to support `env_mat_r` in the future? The current `env_mat` is actually `env_mat_a`.

**Comment by wanghan-iapcm** [comment]
> I still have a question. Do we want to support `env_mat_r` in the future? The current `env_mat` is actually `env_mat_a`.

To compute `env_mat_r`, we can cut the first column of env_mat_a out. The overhead of the extra three columns looks marginal.

**Pull request merged**